biz.aQute.repository/src/aQute/bnd/repository/maven/provider/IndexFile.java

```diff
@@ -338,17 +339,25 @@ private boolean isMulti(String name) {
 	}
 
 	private Map<Archive, Resource> parseMulti(Archive archive, File multi) throws Exception {
+		Set<String> multis = new HashSet<String>();
+		File localArchives = repo.toLocalFile(archive)
+			.getParentFile();
 		Map<Archive, Resource> result = new HashMap<>();
 		try (Jar jar = new Jar(multi)) {
-			int n = 1000;
 			for (Entry<String, aQute.bnd.osgi.Resource> entry : jar.getResources()
 				.entrySet()) {
 				String path = entry.getKey()
 					.toLowerCase();
 				if (path.endsWith(".jar")) {
+					String name = Strings.lastPathSegment(path)[1];
+					name = Strings.extension(name)[0];
```

Comment:
`extension` can return `null`. So you need to guard against NPE here.

Reply:
Same, there must be an extension

Reply:
In this case we don't because we already checked if the name ends with ".jar" in the if above.